tests/tests/pos/surface/final_assoc_reft01.rs

```diff
@@ -0,0 +1,20 @@
+#[flux::assoc(fn f1() -> bool)]
+trait Baz {
+    #[flux::spec(fn (s: Self) -> bool[<Self as Baz>::f1()])]
+    fn into_bool(self) -> bool;
+}
+
+#[flux::final_assoc(fn f2() -> bool { <T as Baz>::f1() })]
```

Comment:
we could make this `#[flux::assoc(final fn f2() -> bool { <T as Baz>::f1() })`. There is a proposal to do the same for trait methods https://internals.rust-lang.org/t/pre-rfc-final-trait-methods/18407, so we could be consistent with that. But I don't have strong opinions.